### PR TITLE
Create 2021-03-23-solidity-0.8.3-release-announcement.md

### PR DESCRIPTION
**Pull request by franzihei**

### CODE REVIEW

File: _posts/2021-03-23-solidity-0.8.3-release-announcement.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+---
+layout: post
+published: true
+title: 'Solidity 0.8.3 Release Announcement'
+date: '2021-03-23'
+author: Solidity Team
+category: Releases
+---
+
+[Solidity v0.8.3](https://github.com/ethereum/solidity/releases/tag/v0.8.3) fixes the [Solidity Optimizer Keccak Caching Bug](https://blog.soliditylang.org/2021/03/23/keccak-optimizer-bug/), which is present in all prior versions of Solidity, and includes two changes to the optimizer (improve storage writes of small types and replace multiplications and divisions with powers of two by shifts).
```
Comment (chriseth) [inline _posts/2021-03-23-solidity-0.8.3-release-announcement.md:10]:
```suggestion
[Solidity v0.8.3](https://github.com/ethereum/solidity/releases/tag/v0.8.3) fixes the [Solidity Optimizer Keccak Caching Bug](https://blog.soliditylang.org/2021/03/23/keccak-optimizer-bug/), which is present in all prior versions of Solidity, and, in addition, includes two improvements to the optimizer (improve storage writes of small types and replace multiplications and divisions with powers of two by shifts).
```

File: _posts/2021-03-23-solidity-0.8.3-release-announcement.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
```
Comment (chriseth) [inline _posts/2021-03-23-solidity-0.8.3-release-announcement.md:11]: We could mention that the improvements should provide a big gas benefit when writing structs that cover a full storage slot to storage.

Comment (hrkrshnn) [inline _posts/2021-03-23-solidity-0.8.3-release-announcement.md:11]: The improvement will make writing packed structs more gas efficient.  For example, in the following example, writing to the struct avoids reading from storage, i.e., the optimizer would be able to remove the redundant `sload`:

```solidity
struct S {
    address a;
    uint96 b;
}

contract C {
    S s;
    function f(uint96 _b) public {
        s.a = msg.sender;
        s.b = _b;
    }
}
```


-----

In the EVM, the shift operations (`shl`, `shr`, `sar`) are cheaper than the arithmetic operations `mul` and `div`. We added an optimization rule that converts multiplication and division by powers of two into `shl` and `shr` respectively. On top of the small gas saving from this simple rule, the rule also chains well with other shift operations. For example:  `shl(10, mul(x, 256)) -> shl(10, shl(8, x)) -> shl(18, x)`.

Comment (franzihei) [reply _posts/2021-03-23-solidity-0.8.3-release-announcement.md:11]: added

Comment (franzihei) [reply _posts/2021-03-23-solidity-0.8.3-release-announcement.md:11]: added this

File: _posts/2021-03-23-solidity-0.8.3-release-announcement.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+### Solidity Optimizer Keccak Caching Bug
+
+On March 20, 2021, a bug in Solidity’s bytecode optimizer was found by differential fuzzing. The bug is fixed with this version. The bug is present in all prior versions of Solidity and has a severity level of "medium". To learn more about the bug and check if your code is affected by it, please read the corresponding [Security Alert](https://blog.soliditylang.org/2021/03/23/keccak-optimizer-bug/).
+
+## Notable New Features
+
+### Optimizer Improvements
+
+The optimizer improvements included in this release, namely improving storage writes of small types and replacing multiplications and divisions with powers of two by shifts, will make writing packed structs more gas efficient. In the following example, writing to the struct avoids reading from storage, i.e., the optimizer would be able to remove the redundant `sload`:
```
Comment (hrkrshnn) [inline _posts/2021-03-23-solidity-0.8.3-release-announcement.md:22]:
```suggestion
There are two optimizer improvements included in this release, namely improving storage writes for small types and replacing multiplications and divisions with powers of two by shifts. The former will make writing packed structs more gas efficient. In the following example, writing to the struct avoids reading from storage, i.e., the optimizer would be able to remove the redundant `sload`:
```

**Review by hrkrshnn — APPROVED** [review]
Would be nice to get one more approval.

File: _posts/2021-03-23-solidity-0.8.3-release-announcement.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+## Notable New Features
+
+### Optimizer Improvements
+
+There are two optimizer improvements included in this release, namely improving storage writes for small types and replacing multiplications and divisions with powers of two by shifts. The former will make writing packed structs more gas efficient. In the following example, writing to the struct avoids reading from storage, i.e., the optimizer would be able to remove the redundant `sload`:
+
+```solidity
+struct S {
+    address a;
+    uint96 b;
```
Comment (chriseth) [inline _posts/2021-03-23-solidity-0.8.3-release-announcement.md:27]: Don't we need uint136?

Comment (hrkrshnn) [reply _posts/2021-03-23-solidity-0.8.3-release-announcement.md:27]: Just `160 + 96 = 256`. I think this was even the nick johnson's example from twitter.

Comment (chriseth) [reply _posts/2021-03-23-solidity-0.8.3-release-announcement.md:27]: Sorry, my bad. This is correct.

**Review by chriseth — APPROVED** [review]

**Pull request merged**